xapian-letor/api/featurelist.cc

```diff
@@ -100,11 +103,14 @@ FeatureList::create_feature_vectors(const Xapian::MSet & mset, const Xapian::Que
     for (Xapian::MSetIterator i = mset.begin(); i != mset.end(); ++i) {
 	Xapian::Document doc = i.get_document();
 	std::vector<double> fVals;
-
+	// All stats are computed and stored in the corresponding variables.
+	internal->compute_statistics(letor_query, letor_db, doc);
 	for (std::vector<Feature*>::const_iterator it = feature.begin(); it != feature.end(); ++it) {
 	    (*it)->set_database(letor_db);
 	    (*it)->set_query(letor_query);
 	    (*it)->set_doc(doc);
+	    // Populates the Feature with required stats
+	    internal->populate_feature(*it);
 	    vector<double> values = (*it)->get_values();
 	    fVals.insert(fVals.end(), values.begin(), values.end()); // Append feature values
```

Comment:
The trailing comment makes this line too long (Xapian code avoids lines longer than 80 columns).

In general, I'd suggest putting the comment on the line before anyway - trailing comments are much easier to miss when scanning down through the code.